test/test.nodriver.js

```diff
@@ -55,29 +53,73 @@ describe('When No Drivers Are Available', function() {
             });
         });
 
-        it('fails to setDriver ' + driverName + ' [promise]', function(done) {
-            localforage.setDriver(driverName).then(null, function(err) {
+        it('fails to setDriver ' + driverName + ' [promise]', function() {
+            return localforage.setDriver(driverName).then(null, function(err) {
                 expect(err).to.be.an(Error);
                 expect(err.message).to.be('No available storage method found.');
-                done();
             });
         });
     });
 
-    it('fails to setDriver using array parameter [callback]', function(done) {
-        localforage.setDriver(DRIVERS, null, function(err) {
-            expect(err).to.be.an(Error);
-            expect(err.message).to.be('No available storage method found.');
-            done();
+    describe('using array parameter', function() {
+        it('fails to setDriver [callback]', function(done) {
+            localforage.setDriver(STORAGE_DRIVERS, null, function(err) {
+                expect(err).to.be.an(Error);
+                expect(err.message).to.be('No available storage method found.');
+                done();
+            });
+        });
+
+        it('fails to setDriver [promise]', function() {
+            return localforage.setDriver(STORAGE_DRIVERS).then(null, function(err) {
+                expect(err).to.be.an(Error);
+                expect(err.message).to.be('No available storage method found.');
+            });
         });
     });
 
-    it('fails to setDriver using array parameter [promise]', function(done) {
-        localforage.setDriver(DRIVERS).then(null, function(err) {
-            expect(err).to.be.an(Error);
-            expect(err.message).to.be('No available storage method found.');
-            done();
+    describe('when not using memory storage driver', function() {
+
+        before(function() {
+            this.defaultConfig = localforage.config();
+            this.defaultConfig.driver = STORAGE_DRIVERS.slice();
+        });
+
+        // Reset localForage before each test so we can call `config()` without
+        // errors.
+        beforeEach(function() {
+            localforage._ready = null;
+            localforage.config(this.defaultConfig);
+        });
+
+        after(function() {
+            this.defaultConfig.driver = DRIVERS.slice();
```

Comment:
I'm guessing this is to return a new copy, but could we use `[ …DRIVERS ]` for that instead (I forget if ES6 will let us)? Or even use a `deep-copy` library for the tests?  It just seems more idiomatic than using `.slice()`, which to me is more taking advantage of the side-effects of `slice()` than intending to use it to actually change the array. 😃

Reply:
I will add a comment for now explaining the purpose of this, but we should definitely add es6 support in our tests. Can't even use `() => {}` right now, which seems frustrating. I actually got bitten by this while updating the the test cases.

Reply:
I have a branch with this in process. I converted the Api tests already to use karma (tests in all browsers), es6, async/await, etc. I’ll get back to it when I’m back from holiday :-)